### PR TITLE
Deprecate AbstractMatrix inputs

### PR DESCRIPTION
**Pull request by heliosdrm**
To deal with #106 and the last comments on #102 

With this, using `julia --depwarn=yes`:

```julia
julia> using RecurrenceAnalysis
julia> x=randn(100, 2);
julia> RecurrenceMatrix(x,0.4)
┌ Warning: `RecurrenceMatrix(x::AbstractMatrix, ε; kwargs...)` is deprecated, use `RecurrenceMatrix(Dataset(x), ε; kwargs...)` instead.
```

But the calculations go as if the matrix had been converted to a `Dataset`. (By default, in Julia 1.5+ the deprecation warning is hidden).

If this is approved, `src/deprecate.jl` could be extended later on for `RecurrenceMatrix{FAN}` etc.

### CODE REVIEW

**Comment by heliosdrm** [comment]
My doubt is: do we want to keep the deprecation warning as it is by default (silent unless you activate the option `depwarn=yes`, which most people won't do), or is it better to make it more prominent, and always throw the warning to instruct users that `Dataset` must be used for multi-dimensional data?

**Comment by Datseris** [comment]
I think the depwarning always has to be thrown. How did you make it "not being thrown by default"? When I use `@deprecate` or `@warn` the are always shown when starting Julia the default way.

File: src/matrices.jl [hunk 1]
```diff
@@ -68,7 +68,7 @@ Objects of type `<:AbstractRecurrenceMatrix` are displayed as a [`recurrenceplot
 The recurrence matrix is a numeric representation of a "recurrence plot" [^1, ^2],
 in the form of a sparse square matrix of Boolean values.
 
-`x` must be a `Vector` or a `Dataset` or a `Matrix` with data points in rows
+`x` must be a `Vector` or a `Dataset`
 (possibly representing and embedded phase space; see [`embed`](@ref)).
```
Comment (Datseris) [inline src/matrices.jl:72]: spotted a typo here: "possibly representing **an** embedded space"

**Comment by heliosdrm** [comment]
> How did you make it "not being thrown by default"? When I use `@deprecate` or `@warn` the are always shown when starting Julia the default way.

I did not make it. It's the default as of Julia 1.5:
https://docs.julialang.org/en/v1/base/base/#Base.@deprecate

**Comment by Datseris** [comment]
Okay then it is an easy change. Instead of using `@deprecate` we can instead use `@warn` inside the function call.

**Comment by heliosdrm** [comment]
`@warn` would repeat the message every time matrices are used as inputs, whereas `@deprecate` would show the message only the first time it happens. Also, according to [the answers in Discourse](https://discourse.julialang.org/t/why-is-depwarn-no-by-default/51094/11) in 1.6 we would be able to force the deprecation message.

Should we make it a regular warning, instead of a deprecation warning, still?

**Comment by Datseris** [comment]
> `@warn` would repeat the message every time matrices are used as inputs

That's what I always do actually. Don't know if it is good practice or not, but if I know there is some code that will break in the next release, I'd rather press the user to change it as soon as possible instead of passively letting it go after 1 warning.

**Pull request merged**